src/main/java/com/greenfox/poker/service/ShowDownService.java

```diff
@@ -0,0 +1,185 @@
+package com.greenfox.poker.service;
+
+
+import com.greenfox.poker.model.Card;
+import com.greenfox.poker.model.GamePlayer;
+import com.greenfox.poker.model.GamePlayerDTO;
+import com.greenfox.poker.model.GameState;
+import com.greenfox.poker.model.ShowDownResult;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ShowDownService {
+
+
+  @Autowired
+  GameService gameService;
+
+  @Autowired
+  GameState gameState;
+
+  private List<Card[]> chosen5Cards = new ArrayList<>();
+
+  public ShowDownResult whoIsTheWinner(long id) {
```

Comment:
This is quite a big function, this could be separated to individual ones that do just one certain subtask.